src/GameActivity.java

```diff
@@ -21,14 +22,20 @@ public static void main(String[] args) throws FileNotFoundException {
         game.data.loadArrayList();
         Collections.shuffle(hintArrayList);
         for (Hint hint : hintArrayList) {
-            System.out.println(hint.getHintText());
-            System.out.println("Please Enter Your Answer");
-            String userAnswer = userInput.nextLine();
-            if (userAnswer.equalsIgnoreCase(hint.getCorrectAnswer())) {
-                playerScore++;
-                System.out.println("Correct!");
-            } else {
-                System.out.println("Sorry, that was incorrect! The correct answer is: " + hint.getCorrectAnswer());
+            boolean isCorrect = false;
+            while (!isCorrect) {
+                System.out.println(hint.getHintText());
+                System.out.println("Please Enter Your Answer");
+                String userAnswer = userInput.nextLine();
+                if (userAnswer.equalsIgnoreCase(hint.getCorrectAnswer())) {
+                    playerScore++;
+                    isCorrect = true;
+                    System.out.println("Correct!");
+                } else {
+                    System.out.println("Sorry, that was incorrect! Try Again? (Y/N)");
+                    String tryAgain = userInput.nextLine();
+                    isCorrect = !tryAgain.equalsIgnoreCase("Y");
```

Comment:
It looks like if the user doesn't select to try again they just move on to the next hint? Let's not do that, they're blocked from going forward.